File: 2015/go/day5/day5p1.go
// package day5 contains the solutions to the Day 5 challenge
// of the 2015 Advent of Code challenge.
package day5

import (
	"../utils"
	"fmt"
	"strings"
)

var vowels = []rune{'a', 'e', 'i', 'o', 'u'}
var badStrings = []string{"ab", "cd", "pq", "xy"}

func Day5Part1() {
	niceStrings := 0
	var input = utils.GetInputFileAsString("day5input.txt")
	var lines = strings.Split(input, "\n")
	for _, line := range lines {
		hasDouble, hasBadString := false, false
		vowelsCount := 0
		var currentChar, lastChar rune
		for _, char := range line {
			currentChar = char
			if lastChar != 0 {
				if lastChar == currentChar {
					hasDouble = true
				}
				charPair := string(lastChar) + string(currentChar)
				for _, bs := range badStrings {
					if bs == charPair {
						hasBadString = true
						break
					}
				}
			}
			if hasBadString {
				break
			}
			for _, vowel := range vowels {
				if char == vowel {
					vowelsCount += 1
				}
			}
			lastChar = currentChar
		}
		if hasDouble && vowelsCount >= 3 && !hasBadString {
			niceStrings++
		}
	}
	fmt.Println(niceStrings)
}

File: 2015/day5/day5p1.py
import sys

file = open("../inputs/day5input.txt", "r")
directions = file.read().split("\n")

vowels = ["a","e","i","o","u"]
badStrings = ["ab", "cd", "pq", "xy"]

goodStrings = 0
for d in directions:
	#bad string check
	hasBadString = False
	for bs in badStrings:
		if bs in d:
			hasBadString = True
	if hasBadString:
		continue
	#double check
	hasDouble = False
	for x in range(0, len(d)-1):
		if d[x] == d[x+1]:
			hasDouble = True
	if not hasDouble:
		continue
	#vowel check
	numVowels = 0
	for v in vowels:
		numVowels += d.count(v)

	if numVowels < 3:
		continue

	goodStrings += 1
	
print(str(goodStrings))

